File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    # 에라토스테네스의 체를 활용
    
    # 0 ~ n까지의 모든 수를 우선 소수로 가정하여 True로 채운 배열을 선언한다
    tempArr = [True] * (n + 1)
    # 0과 1은 소수가 아니기 때문에 False를 대입해준다
    tempArr[0], tempArr[1] = False, False
    
    for i in range(2, int(n**0.5) + 1):
        # 만약 i가 소수라면
        if tempArr[i]:
            # i의 배수들은 소수가 아니므로 False로 바꿔준다.
            for j in range(i * i, n + 1, i):
                tempArr[j] = False

    return sum(tempArr)

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    var tempArr = [];
    
    // 배열에 boolean값을 넣어준다
    for (let i = 0; i <= n; i++) {
        tempArr.push(true);
    }
    
    // 0과 1은 소수가 아니므로 false를 넣어준다
    tempArr[0] = false;
    tempArr[1] = false;
    
    // 숫자 2부터 제곱근까지 탐색하면서
    for (let i = 2; i <= Math.floor(n**0.5) + 1; i++) {
        // 만약 i가 소수라면
        if (tempArr[i]) {
            // i의 배수들은 전부 false로 처리
            for (let j = i * i; j <= n; j += i) {
                tempArr[j] = false;
            }
        }
    }
    
    // tempArr 배열에서 true의 수를 reduce 함수를 통해 계산
    var answer = tempArr.reduce((a, b) => a + b, 0);

    return answer;
}